engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -1716,6 +1720,22 @@ struct dark_thought_t final : public priest_buff_t<buff_t>
       }
     }
 
+    // TODO: check if you can have multiple out at once
+    if ( priest().sets->has_set_bonus( PRIEST_SHADOW, T28, B4 ) )
```

Comment:
This might be a micro-optimization, but you could store this in a boolean which is set in the constructor.